test/meson.build

```diff
@@ -225,33 +249,39 @@ etl_test_sources = files(
     'test_utility.cpp',
     'test_variance.cpp',
     'test_variant_legacy.cpp',
-    'test_variant_pool.cpp',
     'test_variant_pool_external_buffer.cpp',
+    'test_variant_pool.cpp',
     'test_variant_variadic.cpp',
-    'test_vector.cpp',
     'test_vector_external_buffer.cpp',
     'test_vector_non_trivial.cpp',
-    'test_vector_pointer.cpp',
     'test_vector_pointer_external_buffer.cpp',
+    'test_vector_pointer.cpp',
+    'test_vector.cpp',
     'test_visitor.cpp',
     'test_xor_checksum.cpp',
-    'test_xor_rotate_checksum.cpp',
+    'test_xor_rotate_checksum.cpp'
 )
 
 compile_args = [
     '-DENABLE_ETL_UNIT_TESTS',
     '-DETL_DEBUG',
 ]
+link_args = []
 
 if get_option('use_stl')
     compile_args += '-DETL_NO_STL=0'
 elif
     compile_args += '-DETL_NO_STL=1'
 endif
 
-if meson.get_compiler('cpp').get_id() == 'gcc'
-    etl_test_sources += files('test_atomic_gcc_sync.cpp')
+if meson.get_compiler('cpp').get_argument_syntax() == 'gcc'
+    compile_args += '-fsanitize=address,undefined'
     compile_args += '-fexceptions'
+    compile_args += '-Wall'
+    compile_args += '-Wextra'    
+    compile_args += '-Wno-non-virtual-dtor' #TODO remove and fix warning in code
+    compile_args += '-Werror'
+    link_args += '-fsanitize=address,undefined'
```

Comment:
If it's not good enough to set those via `meson setup -DXXX=...` then you can force it per target with: 

```meson
executable(
    ...,
    override_options: [
        'werror=true',
        'warning_level=3',
        'b_sanitize=address,undefined'
    ],
)
```

Reply:
I have added thises lines to be in sync with the CMake build

Reply:
Do I need to change the code ?